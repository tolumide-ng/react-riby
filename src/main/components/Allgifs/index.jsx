import React, { useState, useEffect } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { fetchAllGifAction } from '../../../store/modules/allGifs/actions';
import SingleGif from '../SingleGif/index.jsx';
import Loader from '../Loader';

const AllGifs = ({
  fetchAllGifs,
  history,
  allGifs,
  allGifStatus,
  allGifError,
}) => {
  useEffect(() => {
    fetchAllGifs({ history });
  }, []);
  return (
    <div className="w-full flex flex-row flex-wrap mx-auto justify-center">
      {allGifStatus === 'pending' && (
       <div className='mt-20'> <Loader /> </div>
      )}
      {allGifStatus === 'success' &&
        allGifs.map(gif => (
          <SingleGif
            imgSrc={gif.images.original.url}
            title={gif.title}
            key={gif.slug}
            id={gif.id}
          />
        ))}
    </div>
  );
};

const mapStateToProps = state => ({
  allGifs: state.allGifReducer.allGif,
  allGifStatus: state.allGifReducer.allGifStatus,
  allGifError: state.allGifReducer.allGifError,
});

const mapDispatchToProps = dispatch => ({
  fetchAllGifs: ({ history }) =>
    dispatch(fetchAllGifAction({ history })),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(withRouter(AllGifs));
