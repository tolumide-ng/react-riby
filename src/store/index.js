import { createStore, applyMiddleware } from 'redux';
import ReduxThunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import rootReducer from './modules';

const middlewares = applyMiddleware(ReduxThunk);

const store = createStore(rootReducer, composeWithDevTools);

export default store;
