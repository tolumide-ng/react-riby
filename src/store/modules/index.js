import { combineReducers } from 'redux';
import allGifReducer from './allGifs/reducers';
import detailedGifReducer from './detailedGif/reducers';
import searchGifReducer from './searchGif/reducers';

const rootReducer = combineReducers({
  allGifReducer,
  detailedGifReducer,
  searchGifReducer,
});

export default rootReducer;
