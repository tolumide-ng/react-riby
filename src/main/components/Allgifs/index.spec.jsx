/* eslint-disable react/jsx-props-no-spreading */
import { render, fireEvent, cleanup } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import Allgifs from './index';
import store from '../../../store';

const history = { push: jest.fn() };
const fetchAllGifs = jest.fn(() => {
  return [
    {
      id: '329y3249',
      title: 'title',
      source: 'source',
      rating: 'rating',
      username: 'username',
      images: {
        original: {
          url: 'checkout',
        },
      },
    },
    {
      id: '45',
      title: 'title',
      source: 'source',
      rating: 'rating',
      username: 'username',
      images: {
        original: {
          url: 'anotherone',
        },
      },
    },
  ];
});

const allGifStatus = '';

describe('AllGifs component', () => {
  test('Should render the allGifs component', () => {
    const defaultProps = {
      history,
      allGifStatus,
      fetchAllGifs,
    };
    const { asFragment } = render(
      <Provider store={store}>
        <MemoryRouter>
          <Allgifs {...defaultProps} />
        </MemoryRouter>
      </Provider>,
    );
    expect(asFragment).toMatchSnapshot();
  });

  test('Should render loader when allGifStatus is pending', () => {
    const defaultProps = {
      history,
      allGifStatus: 'pending',
      fetchAllGifs,
    };
    const { asFragment, getByTestId } = render(
      <Provider store={store}>
        <MemoryRouter>
          <Allgifs {...defaultProps} />
        </MemoryRouter>
      </Provider>,
    );
    expect(asFragment).toMatchSnapshot();
    expect(getByTestId('pageloading')).toBeTruthy();
  });
});
