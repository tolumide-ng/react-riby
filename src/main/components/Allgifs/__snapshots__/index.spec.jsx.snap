// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllGifs component Should render loader when allGifStatus is pending 1`] = `[Function]`;

exports[`AllGifs component Should render the allGifs component 1`] = `[Function]`;
